drivers/at86rf215/at86rf215_netdev.c

```diff
@@ -601,11 +645,88 @@ static int _set(netdev_t *netdev, netopt_t opt, const void *val, size_t len)
                 res = sizeof(uint8_t);
                 break;
 #endif /* MODULE_NETDEV_IEEE802154_MR_OFDM */
+#ifdef MODULE_NETDEV_IEEE802154_MR_FSK
+            case IEEE802154_PHY_MR_FSK:
+                at86rf215_configure_FSK(dev,
+                                        at86rf215_FSK_get_srate(dev),
+                                        at86rf215_FSK_get_mod_idx(dev),
+                                        at86rf215_FSK_get_mod_order(dev),
+                                        at86rf215_FSK_get_fec(dev));
+                res = sizeof(uint8_t);
+                break;
+#endif /* MODULE_NETDEV_IEEE802154_MR_FSK */
             default:
                 return -ENOTSUP;
             }
             break;
 
+#ifdef MODULE_NETDEV_IEEE802154_MR_FSK
+        case NETOPT_MR_FSK_MODULATION_INDEX:
+            if (at86rf215_get_phy_mode(dev) != IEEE802154_PHY_MR_FSK) {
+                return -ENOTSUP;
+            }
+
+            if (at86rf215_FSK_set_mod_idx(dev, *(uint8_t *)val) == 0) {
+                res = at86rf215_FSK_get_mod_idx(dev);
+            } else {
+                res = -ERANGE;
+            }
+            break;
+
+        case NETOPT_MR_FSK_MODULATION_ORDER:
+            if (at86rf215_get_phy_mode(dev) != IEEE802154_PHY_MR_FSK) {
+                return -ENOTSUP;
+            }
+
+            if (*(uint8_t *)val != 2 && *(uint8_t *)val != 4) {
+                res = -ERANGE;
+            } else {
+                at86rf215_FSK_set_mod_order(dev, *(uint8_t *)val >> 2);
+                res = sizeof(uint8_t);
+            }
+            break;
+
+        case NETOPT_MR_FSK_SRATE:
+            if (at86rf215_get_phy_mode(dev) != IEEE802154_PHY_MR_FSK) {
+                return -ENOTSUP;
+            }
+
+            res = _get_best_match(at86rf215_fsk_srate_10kHz,
+                                  FSK_SRATE_400K + 1, *(uint16_t *)val / 10);
+            if (at86rf215_FSK_set_srate(dev, res) == 0) {
```

Comment:
Magic numbers?

Reply:
added a comment.
(Not sure if it makes sense to use 2 and 4 in NETOPT since there is no '3-FSK' or '17-FSK', but it sure makes `ifconfig` easier that way)